app/weather_forecast.py

```diff
@@ -0,0 +1,265 @@
+from datetime import datetime, timedelta
+from dotenv import load_dotenv
+from os import getenv
+import requests
+
+
+""" This feature requires an API KEY - get yours free @ visual-crossing-weather.p.rapidapi.com """
+
+SUCCESS_STATUS = 0
+ERROR_STATUS = -1
+HISTORY_TYPE = "history"
+HISTORICAL_AVERAGE_TYPE = "historical-average"
+FORECAST_TYPE = "forecast"
+HISTORY_URL = "https://visual-crossing-weather.p.rapidapi.com/history"
+FORECAST_URL = "https://visual-crossing-weather.p.rapidapi.com/forecast"
+HEADERS = {
+    'x-rapidapi-host': "visual-crossing-weather.p.rapidapi.com"
+    }
+BASE_QUERY_STRING = {"aggregateHours": "24", "unitGroup": "metric", "dayStartTime": "00:00:01",
+                     "contentType": "json", "dayEndTime": "23:59:59", "shortColumnNames": "True"}
+HISTORICAL_AVERAGE_NUM_OF_YEARS = 3
+OUTPUT = {"Status": SUCCESS_STATUS, "ErrorDescription": None, "MinTempCel": None, "MaxTempCel": None,
+          "MinTempFar": None, "MaxTempFar": None, "Conditions": None, "ForecastType": None}
+
+
+def validate_date_input(day, month, year):
+    """ date validation.
+    Args:
+        day (int / str) - day part of date.
+        month (int / str) - month part of date.
+        year (int / str) - year part of date.
+    Returns:
+        (bool) - validate ended in success or not.
+        day (int) - day part of date.
+        month (int) - month part of date.
+        year (int) - year part of date.
+    """
+    try:
+        day = int(day)
+        month = int(month)
+        year = int(year)
+    except ValueError:
+        return False, day, month, year
+    if 1975 <= year <= 2050:
+        try:
+            datetime(year=year, month=month, day=day)
+        except ValueError:
+            return False, day, month, year
+    else:
+        return False, day, month, year
+    return True, day, month, year
+
+
+def get_data_from_api(url, input_query_string):
```

Comment:
The name of the function is very general while it's goal is more specific..consider rename.